dimager/views.py

```diff
@@ -1,3 +1,109 @@
 from django.shortcuts import render
+from django.contrib.auth.decorators import login_required
+from dimager.models import ImagerProfile
+from imager_images.models import Album, Photo, User
+from django.views.generic import CreateView, UpdateView, DeleteView
 
-# Create your views here.
+
+@login_required()
+def user_profile(request, id):
+    user = User.objects.get(id=id)
+    try:
+        albums = Album.objects.filter(user=user).all()
+    except:
+        albums = None
+    profile = ImagerProfile.objects.get(user=user)
+    num_photos = Photo.objects.filter(user=user).count()
+    return render(request, 'profile.html', {
+        'albums': albums,
+        'user': user,
+        'profile': profile,
+        'num_photo': num_photos
+    })
+
+
+@login_required()
+def user_profile(request):
+    user = request.user
+    try:
+        albums = Album.objects.filter(user=user).all()
+    except:
+        albums = None
+    profile = ImagerProfile.objects.get(user=user)
+    num_photos = Photo.objects.filter(user=user).count()
+    return render(request, 'profile.html', {
+        'albums': albums,
+        'user': user,
+        'profile': profile,
+        'num_photo': num_photos
+    })
+
+
+@login_required()
+def library(request):
+    user = request.user
+    try:
+        albums = Album.objects.filter(user=user).all()
+    except:
+        albums = None
+    try:
+        photos = Photo.objects.filter(user=user).all()
+    except:
+        photos = None
+    return render(request, 'library.html', {
+        'user': user,
+        'albums': albums,
+        'photos': photos
+    })
+
+
+@login_required()
+def stream(request):
+    user = request.user
+    try:
+        albums = Album.objects.filter(user=user).order_by('-date_created')
+    except:
+        albums = None
```

Comment:
Depending on whether line 64 above raises an exception the value bound to the symbol `albums` will be an iterable or `None`, which is not iterable.  This means that in the context of the template where you build a display of albums, you'll need to give a shit what type the value is.  That's no good.  

Can you think of a value you could use in the except clause to bind to `albums` that would still be an iterable?